src/ape/managers/chain.py

```diff
@@ -276,13 +282,16 @@ def poll_blocks(
             raise ValueError("'stop' argument must be in the future.")
 
         # Get number of last block with the necessary amount of confirmations.
-        latest_confirmed_block_number = self.height - required_confirmations
-        has_yielded_before_reorg = False
+        last_yielded_hash = None
+        last_yielded_number = None
 
         if start_block is not None:
-            # Front-load historically confirmed blocks.
-            yield from self.range(start_block, latest_confirmed_block_number + 1)
-            has_yielded_before_reorg = True
+            # Front-load historical blocks.
```

Comment:
Or another option
```suggestion
            last_yielded_number = self.height - required_confirmations
            yield from self.range(start_block, last_yielded_number + 1)
            block = self[last_yielded_number + 1]
            last_yielded_hash = block.hash
            last_yielded_number = block.number
            time_since_last = time.time()
            yield block
```

Reply:
I think there could be a race condition in this one.
I like the idea of using `block` from the outer scope because than it for sure is the block that got yielded